apps/passport-client/components/modals/RequireAddPasswordModal.tsx

```diff
@@ -0,0 +1,69 @@
+import { useCallback, useState } from "react";
+import styled from "styled-components";
+import { useDispatch, useSelf } from "../../src/appHooks";
+import { setPassword } from "../../src/password";
+import { BigInput, H2, Spacer } from "../core";
+import { NewPasswordForm } from "../shared/NewPasswordForm";
+
+/**
+ * This uncloseable modal is shown to users of Zupass who have a sync key,
+ * and have never created a password. It asks them to create a password.
+ */
+export function RequireAddPasswordModal() {
+  const [loading, setLoading] = useState(false);
+  const dispatch = useDispatch();
+  const self = useSelf();
+
+  const [newPassword, setNewPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [revealPassword, setRevealPassword] = useState(false);
+  const [error, setError] = useState<string | undefined>();
+
+  const onAddPassword = useCallback(async () => {
+    if (loading) return;
+    setLoading(true);
```

Comment:
can you make the UI do something different while in a loading state?

you can change this value in local development to simulate a slower-than-development connection (which is typically instant). It is really important that all of our interactions feel snappy. and not confusing.

https://github.com/proofcarryingdata/zupass/blob/main/packages/passport-interface/src/api/makeRequest.ts#L71-L76

I recommend setting that value and going through the login flow to see how I've handled the loading states.